diff --git a/ngu/hash.c b/ngu/hash.c
index 72a16d5..c8e0653 100644
--- a/ngu/hash.c
+++ b/ngu/hash.c
@@ -286,7 +286,7 @@ void ripemd160(const uint8_t *msg, int msglen, uint8_t digest[20])
 #error "untested; suspect endian challenge here"
 #endif
 
-    if(((uint32_t)digest) & 0x3) {
+    if(((uintptr_t)digest) & 0x3) {
         // unaligned case
         uint32_t    ctx[5];
 
diff --git a/ngu/random.c b/ngu/random.c
index 4bd3d8d..deef1d2 100644
--- a/ngu/random.c
+++ b/ngu/random.c
@@ -32,7 +32,7 @@ extern uint32_t rng_get(void);
 
 #ifdef UNIX
 # define CHIP_TRNG_SETUP()      
-# define CHIP_TRNG_32()         arc4random()
+# define CHIP_TRNG_32()         random()
 #endif
 
 #ifndef CHIP_TRNG_SETUP
